CMakeLists.txt

```diff
@@ -282,6 +282,32 @@ macro (fortran_hook)
 endmacro (fortran_hook)
 
 macro (files_hook)
+  if(MPI_FOUND AND HDF5_FOUND AND NOT HDF5_IS_PARALLEL)
+    message(WARNING "When building parallel OPM flow we need a "
+      "parallel version of hdf5, but found only a serial one. "
+      "Please install a parallel hdf5 library for MPI "
+      "(e.g with apt-get install libhdf5-mpi-dev) and do a clean "
+      "rebuild (build after \"make clean\". Continuing with "
+      "only normal restart (not hdf5 enabled one).")
+    set(HDF5_FOUND OFF)
+    unset(HAVE_HDF5)
+  endif()
+
+  # CMakeLists_files.cmake is already processed before this hook.
+  # Hence we need to add all HDF5 specific files here to skip them
+  # if we do not find a parallel HDF5 in a parallel build.
+  if(HDF5_FOUND)
+    list(APPEND MAIN_SOURCE_FILES opm/simulators/utils/HDF5File.cpp)
+    list(APPEND TEST_SOURCE_FILES tests/test_HDF5File.cpp)
+    list(APPEND TEST_SOURCE_FILES tests/test_HDF5Serializer.cpp)
```

Comment:
Since you're moving these anyway, is there a reason not to coalesce the requests, i.e.,
```cmake
list(APPEND TEST_SOURCE_FILES
  tests/test_HDF5File.cpp
  tests/test_HDFSerializer.cpp
)
```
instead?

Reply:
Not an issue anymore as it stays as it was before PR: